mlflow/utils/environment.py

```diff
@@ -662,7 +643,7 @@ def _validate_version_constraints(requirements):
 
     try:
         subprocess.run(
-            ["pip", "install", "--dry-run", "-r", tmp_file_name],
+            [sys.executable, "-m", "pip", "install", "--dry-run", "-r", tmp_file_name],
```

Comment:
This is added to ensure that we're running the exact same python version for the pip call to prevent unexpected issues with version resolution, depending on how the Python default loader is configured.